tests/pytests/unit/states/test_docker_volume.py

```diff
@@ -0,0 +1,171 @@
+"""
+Unit tests for the docker state
+"""
+
+import pytest
+import salt.modules.dockermod as docker_mod
+import salt.states.docker_volume as docker_state
+from tests.support.mock import Mock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {
+        docker_mod: {"__context__": {"docker.docker_version": ""}},
+        docker_state: {"__opts__": {"test": False}},
+    }
+
+
+def test_present():
+    """
+    Test docker_volume.present
+    """
+    volumes = []
+    default_driver = "dummy_default"
+
+    def create_volume(name, driver=None, driver_opts=None):
+        for v in volumes:
+            assert v["Name"] != name
+        if driver is None:
+            driver = default_driver
+        new = {"Name": name, "Driver": driver}
+        volumes.append(new)
+        return new
+
+    def remove_volume(name):
+        removed = [v for v in volumes if v["Name"] == name]
+        assert 1 == len(removed)
+        volumes.remove(removed[0])
+        return removed[0]
+
+    docker_create_volume = Mock(side_effect=create_volume)
+    __salt__ = {
+        "docker.create_volume": docker_create_volume,
+        "docker.volumes": Mock(return_value={"Volumes": volumes}),
+        "docker.remove_volume": Mock(side_effect=remove_volume),
+    }
+    with patch.dict(docker_state.__dict__, {"__salt__": __salt__}):
+        ret = docker_state.present("volume_foo")
+        docker_create_volume.assert_called_with(
+            "volume_foo", driver=None, driver_opts=None
+        )
+        assert {
+            "name": "volume_foo",
+            "comment": "",
+            "changes": {"created": {"Driver": default_driver, "Name": "volume_foo"}},
+            "result": True,
+        } == ret
+        assert len(volumes) == 1
+        assert volumes[0]["Name"] == "volume_foo"
+        assert volumes[0]["Driver"] is default_driver
+
+        # run it again with the same arguments
+        orig_volumes = [volumes[0].copy()]
+        ret = docker_state.present("volume_foo")
+        assert {
+            "name": "volume_foo",
+            "comment": "Volume 'volume_foo' already exists.",
+            "changes": {},
+            "result": True,
+        } == ret
+        assert orig_volumes == volumes
+
+        # run it again with a different driver but don't force
+        ret = docker_state.present("volume_foo", driver="local")
+        assert {
+            "name": "volume_foo",
+            "comment": (
+                "Driver for existing volume 'volume_foo'"
+                " ('dummy_default') does not match specified"
+                " driver ('local') and force is False"
+            ),
+            "changes": {},
+            "result": False,
+        } == ret
```

Comment:
```suggestion
        assert ret == {
            "name": "volume_foo",
            "comment": (
                "Driver for existing volume 'volume_foo'"
                " ('dummy_default') does not match specified"
                " driver ('local') and force is False"
            ),
            "changes": {},
            "result": False,
        }
```
this makes more sense on failures when pytest shows us a diff, it should always be `what_we_got == what_we_expect`

Reply:
yeah right, I missed it somehow, fixed now